feature-flag/src/main/kotlin/com/linecorp/android/featureflag/FeatureFlagTask.kt

```diff
@@ -125,9 +125,23 @@ abstract class FeatureFlagTask : DefaultTask() {
         buildEnvironment: BuildEnvironment
     ): FeatureFlagData {
         val options = FeatureFlagOptionParser.parse(entry.option)
-        val selectors = FeatureFlagSelectorParser.parse(entry.value)
-        val evaluatedSelectors = FeatureFlagSelectorEvaluator.evaluate(selectors, buildEnvironment)
-        val value = FeatureFlagValueOptimizer.optimize(evaluatedSelectors)
+        val value = convertToFeatureFlagValue(entry.name, entry.value, buildEnvironment)
         return FeatureFlagData(entry.name, value, options)
     }
+
+    private fun convertToFeatureFlagValue(
```

Comment:
We don't use `rawValue` to filter overridden flags while we don't use `flagName` to evaluate flag value.
The simplified structure is as follows:
```kotlin
fun function(valueA: ValueA, valueB: ValueB) = when {
    someCondition -> /* Use `valueA` only */
    someAnotherCondition -> /* Use `valueA` only */
    else -> /* Use `valueB` only */
}
```
In such a case, it might be good to split layers to handle the case. I pasted an example below.

```kotlin
    private fun convertToFeatureFlagData(
        entry: FeatureFlagProperties.Entry,
        buildEnvironment: BuildEnvironment
    ): FeatureFlagData {
        val options = FeatureFlagOptionParser.parse(entry.option)
        val flagName = entry.name
        val value = when (flagName) {
            in forciblyOverriddenFeatureFlags.forciblyEnabledFlags -> FeatureFlagData.Value.True
            in forciblyOverriddenFeatureFlags.forciblyDisabledFlags -> FeatureFlagData.Value.False
            else -> evaluateFlagValue(entry.value, buildEnvironment)
        }
        return FeatureFlagData(flagName, value, options)
    }

    private fun evaluateFlagValue(
        rawValue: String,
        buildEnvironment: BuildEnvironment
    ): FeatureFlagData.Value {
        val selectors = FeatureFlagSelectorParser.parse(rawValue)
        val evaluatedSelectors = FeatureFlagSelectorEvaluator.evaluate(selectors, buildEnvironment)
        return FeatureFlagValueOptimizer.optimize(evaluatedSelectors)
    }
```